File: typescript/Employee/Employee.ts
export class Employee {
    empName: string
    empId: number

    constructor(empName: string, empId: number) {
        this.empName = empName;
        this.empId = empId;
    }

    display = () => console.log(`Employee Name is ${this.empName} and employee ID is ${this.empId}`);
    
}

File: corejava/demo-app/src/com/techlab/demo/Assignment/employee-salary-app/src/com/techlab/empsalary/Employee.java
package com.techlab.empsalary;

public abstract class Employee {

	private int empId;
	private String empName;
	protected float basicSalary;
	
	public Employee(int empId, String empName, float basicSalary) {
		
		this.empId = empId;
		this.empName = empName;
		this.basicSalary = basicSalary;
	}
	
	public abstract double calculateSalary();
	
	public int getId() {
		return empId;
	}
	
	public String getName() {
		return empName;
	}
	
	public float getBasicSalary() {
		return basicSalary;
	}
	
}
